src/btree/bt_sync.c

```diff
@@ -385,10 +385,15 @@ __sync_page_skip(
         return (0);
 
     /*
-     * Skip reading the pages that are normal leaf pages or don't have an aggregated durable stop
-     * timestamp.
+     * Skip reading the on-disk page when it meets any of the following conditions.
+     * 1. Leaf pages with no overflow keys/values.
+     * 2. Internal pages from non-logged tables with no aggregated stop timestamp.
```

Comment:
Could you update this comment to describe why rather than what (or as well as what)?

Seeing WT_ADDR_LEAF_NO, I wonder if it could be purposed to address this as well - could it be set if any leaf page has been deleted, as well as if there are overflow records? Not a change for this ticket, probably, but something to consider.

Reply:
How about the following comment?

  ```
  /*
     * The checkpoint cleanup fast deletes the obsolete leaf page by marking it as deleted
     * in the internal page. To achieve this, 
     * 
     * 1. Checkpoint has to read all the internal pages that have obsolete leaf pages.
     *    To limit the reading of number of internal pages, the aggregated stop durable timestamp
     *    is checked except when the table is logged. Logged tables do not use timestamps.
     * 
     * 2. Obsolete leaf pages with overflow keys/values cannot be fast deleted to free
     *    the overflow blocks. Read the page into cache and mark it dirty to remove the
     *    overflow blocks during reconciliation.
     * 
     * FIXME: Read internal pages from non-logged tables when the remove/truncate
     * operation is performed using no timestamp.
     */
```

I didn't clear about the overflow page. I updated the comment to explain it. Please check and let me know.

Reply:
Looks better, thanks Hari